resteasy-client-vertx/src/main/java/org/jboss/resteasy/client/jaxrs/internal/VertxFutureRxInvokerImpl.java

```diff
@@ -0,0 +1,182 @@
+        invocation.setEntity(entity);
+        return invocation;
+    }
+
+    @Override
+    public Future<Response> get() {
+        return method(HttpMethod.GET);
+    }
+
+    @Override
+    public <T> Future<T> get(Class<T> responseType) {
+        return method(HttpMethod.GET, responseType);
+    }
+
+    @Override
+    public <T> Future<T> get(GenericType<T> responseType) {
+        return method(HttpMethod.GET, responseType);
+    }
+
+    @Override
+    public Future<Response> put(Entity<?> entity) {
+        return method(HttpMethod.PUT, entity);
+    }
+
+    @Override
+    public <T> Future<T> put(Entity<?> entity, Class<T> responseType) {
+        return method(HttpMethod.PUT, entity, responseType);
+    }
+
+    @Override
+    public <T> Future<T> put(Entity<?> entity, GenericType<T> responseType) {
+        return method(HttpMethod.PUT, entity, responseType);
+    }
+
+    @Override
+    public Future<Response> post(Entity<?> entity) {
+        return method(HttpMethod.POST, entity);
+    }
+
+    @Override
+    public <T> Future<T> post(Entity<?> entity, Class<T> responseType) {
+        return method(HttpMethod.POST, entity, responseType);
+    }
+
+    @Override
+    public <T> Future<T> post(Entity<?> entity, GenericType<T> responseType) {
+        return method(HttpMethod.POST, entity, responseType);
+    }
+
+    @Override
+    public Future<Response> delete() {
+        return method(HttpMethod.DELETE);
+    }
+
+    @Override
+    public <T> Future<T> delete(Class<T> responseType) {
+        return method(HttpMethod.DELETE, responseType);
+    }
+
+    @Override
+    public <T> Future<T> delete(GenericType<T> responseType) {
+        return method(HttpMethod.DELETE, responseType);
+    }
+
+    @Override
+    public Future<Response> head() {
+        return method(HttpMethod.HEAD);
+    }
+
+    @Override
+    public Future<Response> options() {
+        return method(HttpMethod.OPTIONS);
+    }
+
+    @Override
+    public <T> Future<T> options(Class<T> responseType) {
+        return method(HttpMethod.OPTIONS, responseType);
+    }
+
+    @Override
+    public <T> Future<T> options(GenericType<T> responseType) {
+        return method(HttpMethod.OPTIONS, responseType);
+    }
+
+    @Override
+    public Future<Response> trace() {
+        return method("TRACE");
+    }
+
+    @Override
+    public <T> Future<T> trace(Class<T> responseType) {
+        return method("TRACE", responseType);
+    }
+
+    @Override
+    public <T> Future<T> trace(GenericType<T> responseType) {
+        return method("TRACE", responseType);
+    }
+
+    @Override
+    public Future<Response> method(String name) {
+        Context vertxContext = Vertx.currentContext();
+        if (vertxContext != null) {
+            return Future.fromCompletionStage(createClientInvocation(name, null).submitCF(), vertxContext);
+        } else {
+            return Future.fromCompletionStage(createClientInvocation(name, null).submitCF());
+        }
```

Comment:
@vietj I'm curious if you have thoughts on this implementation. Do you think it's good enough to just wrap the completion stage in a context-aware way, or is it worth rebuilding some of `ClientInvocation` to leverage things like `executeBlocking` or maybe even try to stay on the event loop thread?